pkg/minikube/audit/audit.go

```
@@ -51,14 +54,40 @@ func args() string {
 }
 
 // Log details about the executed command.
-func Log(startTime time.Time) {
+func LogCommandStart() (string, error) {
 	if len(os.Args) < 2 || !shouldLog() {
-		return
+		return "", fmt.Errorf("this command should not be logged and len(os.Args) : %v, should be less than 2", len(os.Args))
 	}
-	r := newRow(os.Args[1], args(), userName(), version.GetVersion(), startTime, time.Now())
+	id := uuid.New().String()
+	r := newRow(os.Args[1], args(), userName(), version.GetVersion(), time.Now(), id)
 	if err := appendToLog(r); err != nil {
-		klog.Warning(err)
+		return "", fmt.Errorf("%v", err)
 	}
+	return r.id, nil
+}
+
+func LogCommandEnd(id string) error {
+	if currentLogFile == nil {
+		if err := setLogFile(); err != nil {
+			return fmt.Errorf("failed to set the log file: %v", err)
+		}
+	}
+	var auditLogs []byte
+	_, err := currentLogFile.Read(auditLogs)
+	if err != nil {
+		return fmt.Errorf("%v", err)
+	}
+	var rowSlice []row
+	err = json.Unmarshal(auditLogs, &rowSlice)
+	if err != nil {
+		return fmt.Errorf("%v", err)
+	}
+	for _, v := range rowSlice {
+		if v.id == id {
+			v.endTime = time.Now().Format(constants.TimeFormat)
+		}
+	}
+	return nil
```

We should return an error if the entry is never found.